dockstore-common/src/main/java/io/dockstore/common/PipHelper.java

```diff
@@ -69,4 +73,13 @@ public static Map<String, String> convertPipRequirementsStringToMap(String pipRe
         });
         return pipDepMap;
     }
+
+    /**
+     * Validates a string that represents a semantic version.
+     * @param semVer String representing a semantic version.
+     * @return true if the semantic version is valid else false.
+     */
+    public static boolean validateSemVer(String semVer) {
+        return Pattern.matches(SEM_VER_PATTERN.pattern(), semVer);
```

Comment:
SEM_VER_PATTERN is already the compiled pattern, you're unnecessarily getting the precompiled pattern and recompiling it. Just `return SEM_VER_PATTERN.matches(semVer);` should do it.